File: Baekjoon/Class4/04.11053 가장 긴 증가하는 부분 수열.py
import sys
input = sys.stdin.readline

n = int(input())
a = list(map(int, input().split()))
max_lengths = [1] * n

for i in range(n):
    for j in range(i): #각 도착점의 최대 길이 찾기
        if a[i] > a[j]:
            max_lengths[i] = max(max_lengths[i], max_lengths[j] + 1)

print(max(max_lengths))

File: Baekjoon/Class4/04.11053 가장 긴 증가하는 부분 수열.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const n = Number(input[0]);
const a = input[1].split(" ").map(Number);
const maxLengths = new Array(n).fill(1);

for(let i = 0; i < n; i++)
    for(let j = 0; j < i; j++) // 각 도착점의 최대 길이 찾기
        if(a[i] > a[j])
            maxLengths[i] = Math.max(maxLengths[i], maxLengths[j] + 1);

console.log(Math.max(...maxLengths));